### ISSUE TITLE
Terraform Plan Failing with : no Kafka ACLs were matched

### ISSUE THREAD

**Issue by urspraveen2001** [confluentinc/terraform-provider-confluent#141]
We have used the Confluent Kafka provider and created a API key and ACL to associate with the API key. When the ACL was modified or deleted from the Confluent Cloud via Console or CLI the terraform Plan is failing to refresh(Plan) throwing "Error: error reading Kafka ACLs <Actual ACL resource> no Kafka ACLs were matched".

**Comment by linouk23** [comment]
@urspraveen2001 could you share the version of TF Provider you're using and your Terraform configuration?

**Comment by linouk23** [comment]
Just reread your comment and it seems like the expected result was not to throw an error and suggest you to create this new ACLs.

I identified what piece of code we'll need to update, let me check whether my assumption is accurate.

**Comment by linouk23** [comment]
We merged a fix to our internal repo and it should go live in our next release of TF Provider.

**Comment by linouk23** [comment]
Namely, instead of 
```
confluent_kafka_acl.app-consumer-read-on-topic: Refreshing state... [id=lkc-zgqp8z/TOPIC#orders#LITERAL#User:sa-6o8x93#*#READ#ALLOW]
╷
│ Error: error reading Kafka ACLs: error reading Kafka ACLs "lkc-zgqp8z/TOPIC#orders#LITERAL#User:sa-6o8x93#*#READ#ALLOW": no Kafka ACLs were matched
```
you'll get
```
$ terraform plan
...
confluent_kafka_acl.app-consumer-read-on-topic: Refreshing state... [id=lkc-zgqp8z/TOPIC#orders#LITERAL#User:sa-6o8x93#*#READ#ALLOW]

Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the last "terraform apply":

  # confluent_kafka_acl.app-consumer-read-on-topic has been deleted
  - resource "confluent_kafka_acl" "app-consumer-read-on-topic" {
      - host          = "*" -> null
      - id            = "lkc-zgqp8z/TOPIC#orders#LITERAL#User:sa-6o8x93#*#READ#ALLOW" -> null
      - operation     = "READ" -> null
      - pattern_type  = "LITERAL" -> null
      - permission    = "ALLOW" -> null
      - principal     = "User:sa-6o8x93" -> null
      - resource_name = "orders" -> null
      - resource_type = "TOPIC" -> null
      - rest_endpoint = "https://pkc-11111.us-west-2.aws.confluent.cloud:443" -> null

      - credentials {
          - key    = (sensitive value)
          - secret = (sensitive value)
        }

      - kafka_cluster {
          - id = "lkc-zgqp8z" -> null
        }
    }

Unless you have made equivalent changes to your configuration, or ignored the relevant attributes using ignore_changes, the following plan may include actions to undo or
respond to these changes.

────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # confluent_kafka_acl.app-consumer-read-on-topic will be created
  + resource "confluent_kafka_acl" "app-consumer-read-on-topic" {
      + host          = "*"
      + id            = (known after apply)
      + operation     = "READ"
      + pattern_type  = "LITERAL"
      + permission    = "ALLOW"
      + principal     = "User:sa-6o8x93"
      + resource_name = "orders"
      + resource_type = "TOPIC"
      + rest_endpoint = "https://pkc-11111.us-west-2.aws.confluent.cloud:443"

      + credentials {
          + key    = (sensitive value)
          + secret = (sensitive value)
        }

      + kafka_cluster {
          + id = "lkc-zgqp8z"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Note: You didn't use the -out option to save this plan, so Terraform can't guarantee to take exactly these actions if you run "terraform apply" now.
➜  basic-kafka-acls git:(fix) ✗ terraform apply --auto-approve
╷
│ Warning: Provider development overrides are in effect
│ 
│ The following provider development overrides are set in the CLI configuration:
│  - terraform.confluent.io/confluentinc/confluent in /Users/klinou/work/terraform-provider-confluent-internal/bin/darwin-amd64
│ 
│ The behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible with published releases.
╵
...
confluent_kafka_acl.app-producer-write-on-topic: Refreshing state... [id=lkc-zgqp8z/TOPIC#orders#LITERAL#User:sa-5m6vx8#*#WRITE#ALLOW]

Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the last "terraform apply":

  # confluent_kafka_acl.app-consumer-read-on-topic has been deleted
  - resource "confluent_kafka_acl" "app-consumer-read-on-topic" {
      - host          = "*" -> null
      - id            = "lkc-zgqp8z/TOPIC#orders#LITERAL#User:sa-6o8x93#*#READ#ALLOW" -> null
      - operation     = "READ" -> null
      - pattern_type  = "LITERAL" -> null
      - permission    = "ALLOW" -> null
      - principal     = "User:sa-6o8x93" -> null
      - resource_name = "orders" -> null
      - resource_type = "TOPIC" -> null
      - rest_endpoint = "https://pkc-11111.us-west-2.aws.confluent.cloud:443" -> null

      - credentials {
          - key    = (sensitive value)
          - secret = (sensitive value)
        }

      - kafka_cluster {
          - id = "lkc-zgqp8z" -> null
        }
    }

Unless you have made equivalent changes to your configuration, or ignored the relevant attributes using ignore_changes, the following plan may include actions to undo or
respond to these changes.

────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # confluent_kafka_acl.app-consumer-read-on-topic will be created
  + resource "confluent_kafka_acl" "app-consumer-read-on-topic" {
      + host          = "*"
      + id            = (known after apply)
      + operation     = "READ"
      + pattern_type  = "LITERAL"
      + permission    = "ALLOW"
      + principal     = "User:sa-6o8x93"
      + resource_name = "orders"
      + resource_type = "TOPIC"
      + rest_endpoint = "https://pkc-11111.us-west-2.aws.confluent.cloud:443"

      + credentials {
          + key    = (sensitive value)
          + secret = (sensitive value)
        }

      + kafka_cluster {
          + id = "lkc-zgqp8z"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.
confluent_kafka_acl.app-consumer-read-on-topic: Creating...
confluent_kafka_acl.app-consumer-read-on-topic: Still creating... [10s elapsed]
confluent_kafka_acl.app-consumer-read-on-topic: Creation complete after 12s [id=lkc-zgqp8z/TOPIC#orders#LITERAL#User:sa-6o8x93#*#READ#ALLOW]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
...
$ terraform plan
...
confluent_kafka_acl.app-producer-write-on-topic: Refreshing state... [id=lkc-zgqp8z/TOPIC#orders#LITERAL#User:sa-5m6vx8#*#WRITE#ALLOW]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.
```

**Comment by linouk23** [comment]
@urspraveen2001 we've just released `1.15.0` version of TF Provider that included the fix ⬆️ .
